modules/tensor_mechanics/src/materials/ComputeReducedOrderEigenstrain.C

```diff
@@ -0,0 +1,160 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+
+#include "ComputeReducedOrderEigenstrain.h"
+
+#include "MooseMesh.h"
+
+template <>
+InputParameters
+validParams<ComputeReducedOrderEigenstrain>()
+{
+  InputParameters params = validParams<ComputeEigenstrainBase>();
+  params.addRequiredParam<std::vector<MaterialPropertyName>>(
+      "eigenstrain_names", "List of eigenstrains to be applied in this strain calculation");
```

Comment:
`eigenstrain_name` and `eigenstrain_names` for the output and input eigenstrains is a bit of an unfortunate naming convention (I think you took that from the strain calculator). Maybe `input_eigenstrain_names`?